### ISSUE TITLE
Guzzle Config Options not honored fully

### ISSUE THREAD

**Issue by mallardduck** [spatie/crawler#47]
Whoops, sorry for the initially empty issue report. Here's what I meant to include......

------------

I've started playing with this library recently and noticed a few odd things happening with Guzzle options. From what I'm seeing in Crawler.php there are 2 bugs causing this odd behaviour. I could be wrong on these, but from my debugging the current behaviour is errant. 

## 1. Issue with create method logic

Check out [this line](https://github.com/spatie/crawler/blob/master/src/Crawler.php#L42) and line [40 (right above)](https://github.com/spatie/crawler/blob/master/src/Crawler.php#L40).

Since the function call initialises an empty array no matter what, the null coalesce will trigger as false. The default code provided will never be triggered from what I can see; effectively dead code unless I'm mistaken.

If I'm correct that the default options should still be possible then I suggest this:

```
        $client = new Client($clientOptions ?? [
                RequestOptions::COOKIES => true,
            ]);
```
Be changed to:

```
        $client = new Client( (bool) count($clientOptions) ?  $clientOptions : [
                RequestOptions::COOKIES => true,
            ]);
```

## 2. With the update to use Pools none of the client options will ever be used

Since all crawls now use the pool system for Guzzle this means that on [line 122](https://github.com/spatie/crawler/blob/master/src/Crawler.php#L122) we override any user input for options.

Unless there's a reason for this to be done, I would suggest this be updated to use:
> $this->client->getOptions()

...to populate pool options. This way the library will honour either the default settings, or user input, all the way through it's usage of guzzle.

## 3. (Optional) If both are to be addressed...

...then the client creation may need to look like:

```
        $client = new Client( (bool) count($clientOptions) ?  $clientOptions : [
                RequestOptions::CONNECT_TIMEOUT => 10,
                RequestOptions::TIMEOUT => 10,
                RequestOptions::ALLOW_REDIRECTS => false,
                RequestOptions::COOKIES => true,
            ]);
```

-----

So I'm looking forward to some feedback on this; I would have submitted a PR but I'm not 100% sure what is intended here. Once it's been discussed I'd be happy to submit anything as a PR.

**Comment by freekmurze** [comment]
You're right that the passing of options seems to been have coded up a bit sloppy. Sorry for that.

Both issues should be addressed. I welcome a PR that fixes this.

**Comment by mallardduck** [comment]
Awesome, just wanted to make sure I was on the right track with my assumptions!
I can definitely submit a PR to address all that.

To be honest, I've kinda ended up chopping this up for my own needs. In my use case I'd like to track the actual DOM element the links come from, then when results are presented I can present both the link that redirected, the HTML/DOM element that caused it, and an updated element to prevent redirects.

It's still pretty rough, but the current results look something like:

```
SEO Warning: Redirect found 
This HTML link: <a href="/cloud.php">test</a> redirects to: https://www.liquidweb.com/cloudsites/
The final page/URL is: https://www.liquidweb.com/cloudsites/. The corrected link is:<a href="https://www.liquidweb.com/cloudsites/">test</a>
```

In order to accomplish that I had to diverge quite a bit from how things are currently done. Keeping track of links/URLs based on a new class called HtmlLink that tracks the URL and the HtmlNode.

Would that be something you'd be interested in merging as I polish it up and refactor a bit? I know this may only seem like a benefit in specific cases; with an seo-tool in mind though I think this could be really helpful! Let me know your thoughts on that?

I'm really excited and curious to see where we can go with these tools!

**Comment by freekmurze** [comment]
For now, I'd like a PR that just fixes the problems with the config options.

**Comment by freekmurze** [comment]
Your fix is included in the newly tagged version `2.0.7`. Thanks again for your work on this.
